src/map/atcommand.c

```diff
@@ -7695,17 +7696,18 @@ ACMD(duel) {
 		return false;
 	}
 
-	if (!duel->checktime(sd)) {
+	diff = duel->checktime(sd);
+	if (diff > 0) {
 		char output[CHAT_SIZE_MAX];
-		// "Duel: You can take part in duel only one time per %d minutes."
-		sprintf(output, msg_fd(fd,356), battle_config.duel_time_interval);
+		// "Duel: You can take part in duel again after %d secconds."
+		sprintf(output, msg_fd(fd,356), diff);
 		clif->message(fd, output);
 		return false;
 	}
 
 	if (message[0]) {
 		if (sscanf(message, "%12u", &maxpl) >= 1) {
-			if (maxpl < 2 || maxpl > 65535) {
+			if (maxpl < 2 || maxpl > MAX_DUEL) {
```

Comment:
I don't know how duel command works, but you've replaced a 65535 with 1024 in this check..
so what is that exactly ?
and MAX_DUEL seem to be the possible active duels server wide.

Reply:
MAX_DUEL is the maximum number of DUEL group allow to create server wide
if somehow all the players (1027++ players ?) makes it own duel, which is highly unlikely
the `duel_create` function just return 0 without any error ...

I think better make another variable like MAX_DUEL_PLAYERS and save it as 65535 ?
since this 65535 actually has a meaning ...

Reply:
create another variable that aren't used in any other places look unnecessary.

anyway this is outdated diff...

Reply:
hope you would consider finishing emistry. great work dont waste :)